Gordon360/Services/HousingService.cs

```diff
@@ -0,0 +1,774 @@
+        /// - third, it inserts each applicant into the applicants table along with the apartment ID so we know
+        /// which application on which they are an applicant
+        ///
+        /// </summary>
+        /// <param name="userID"> The student username of the user who is attempting to save the apartment application (retrieved via authentication token) </param>
+        /// <param name="sess_cde"> The current session code </param>
+        /// <param name="editorID"> The student username of the student who is declared to be the editor of this application (retrieved from the JSON from the front end) </param>
+        /// <param name="apartmentApplicants"> Array of JSON objects providing apartment applicants </param>
+        /// <param name="apartmentChoices"> Array of JSON objects providing apartment hall choices </param>
+        /// <returns>Returns the application ID number if all the queries succeeded</returns>
+        public int SaveApplication(string userID, string sess_cde, string editorID, ApartmentApplicantViewModel[] apartmentApplicants, ApartmentChoiceViewModel[] apartmentChoices)
+        {
+            CCTEntities1 context = new CCTEntities1();
+
+            IEnumerable<ApartmentAppIDViewModel> idResult = null;
+
+            DateTime now = System.DateTime.Now;
+
+            SqlParameter sessionParam = new SqlParameter("@SESS_CDE", sess_cde);
+            SqlParameter editorParam = new SqlParameter("@STUDENT_ID", editorID);
+
+            // If an application ID was not passed in, then check if an application already exists
+            idResult = RawSqlQuery<ApartmentAppIDViewModel>.query("GET_AA_APPID_BY_STU_ID_AND_SESS @SESS_CDE, @STUDENT_ID", sessionParam, editorParam); //run stored procedure
+            if (idResult != null && idResult.Any())
+            {
+                throw new ResourceCreationException() { ExceptionMessage = "An existing application ID was found for this user. Please use 'EditApplication' to update an existing application." };
+            }
+
+            //----------------
+            // Save the application editor and time
+
+            // All SqlParameters must be remade before being reused in an SQL Query to prevent errors
+            SqlParameter timeParam = new SqlParameter("@NOW", now);
+            editorParam = new SqlParameter("@EDITOR_ID", editorID);
+
+            // If an existing application was not found for this editor, then insert a new application entry in the database
+            int? newAppResult = context.Database.ExecuteSqlCommand("INSERT_AA_APPLICATION @NOW, @EDITOR_ID", timeParam, editorParam); //run stored procedure
+            if (newAppResult == null)
+            {
+                throw new ResourceCreationException() { ExceptionMessage = "The application could not be saved." };
+            }
+
+            //----------------
+            // Retrieve the application ID number of this new application
```

Comment:
You could update your stored procedure to return the generated ID number so that you don't have to do this as two separate queries.